### DISCUSSION TITLE
Test Link with Jest + Testing-Library - "Cannot read property 'push' of null"

### DISCUSSION THREAD

**Discussion by rafaelguedes** [vercel/next.js#22230] [Help]
When I'm testing the `<Link />` behavior, expecting it to redirect to a certain route, there's a TypeError (`Cannot read property 'push' of null`).

```
import React from "react"
import Link from "next/link"

const Sandbox = () => {
  return (
    <div>
      <Link href="/about">
        <a data-testid="mytest">Click Me</a>
      </Link>
    </div>
  )
}

export default Sandbox
```

```
import React from "react"
import { render, fireEvent } from "@testing-library/react"
import { useRouter } from "next/router"
import Sandbox from ".."

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}))

describe("Sandbox", () => {
  it.only("should navigate accordingly", () => {
    const push = jest.fn()
    useRouter.mockImplementationOnce(() => ({
      asPath: "/",
      push,
    }))

    const { getByTestId } = render(<Sandbox />)

    const mytest = getByTestId("mytest")
    fireEvent.click(mytest)
    expect(push).toHaveBeenCalledWith("/about")
  })
})
```

I believe that I've mocked everything I need, so I don't really understand why the router can't actually "push".
What am I missing here?

**Comment by jmhungdev** [comment]
you need to mock `next/link` not `next/router`

**Reply by MHolmes91** [reply]
Is there a good way to mock next/link so that the underlying elements are passed the href? I am using MUI buttons and when the href is not passed to the children, the button renders as a button instead of a link which changes the button's role from a 'link' to a 'button' in React testing library creating an inconsistency in testing vs. in the browser.

I am currently mocking next/link as follows:
```
jest.mock(
  'next/link',
  () =>
    ({ children }: any) =>
      children
)
```

**Reply by mnca16** [reply]
How you implement the test after mocking next/link? I'm also using MUI button, I just started to learn testing and I haven't find a good test example for this case.
